src/plots/mapbox/mapbox.js

```diff
@@ -151,7 +151,7 @@ proto.updateMap = function(calcData, fullLayout, resolve, reject) {
     var promises = [];
     var styleObj = getStyleObj(opts.style);
 
-    if(self.styleObj.id !== styleObj.id) {
```

Comment:
At first I was concerned about order of keys:
```js
JSON.stringify({a: 1, b: 2}) !== JSON.stringify({b: 2, a: 1})
```
But I see we're making `styleObj` in a predictable order in the `getStyleObj` function, so this should be robust 🎉